### PR TITLE
Add customCacheDirectory option

### PR DESCRIPTION
**Pull request by JacobHearst**
## :scroll: Description

This PR adds the ability for developers to specify a custom location for Sentry's cache location. 

## :bulb: Motivation and Context

This option is useful in scenarios where the application hosting Sentry doesn't have read/write permissions for the default cache location. This is particularly relevant for macOS applications running prelogin

## :green_heart: How did you test it?

Built a security agent plugin with the fork, specified the cache location, and ran the plugin at the login window

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Comment by armcknight** [comment]
Thanks for the contribution @JacobHearst ! Just a couple small things I see but overall a nice thing to have 👍🏻

File: Sources/Sentry/Public/SentryOptions.h [hunk 1]
```diff
@@ -16,6 +16,11 @@ NS_SWIFT_NAME(Options)
 - (_Nullable instancetype)initWithDict:(NSDictionary<NSString *, id> *)options
                       didFailWithError:(NSError *_Nullable *_Nullable)error;
 
+/**
+ * Optional, custom cache directory. Use when default one can not be accessed, e.g. in security environment.
+ */
+@property (nonatomic, strong) NSString *_Nullable customCacheDirectory;
```
Comment (philipphofmann) [inline Sources/Sentry/Public/SentryOptions.h:22]:
```suggestion
@property (nonatomic, strong) NSString *_Nullable cacheDirectory;
```

We can set the default in the `SentryOptions.m` to `NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES)`

**Review by philipphofmann — COMMENTED** [review]
Thanks for looking into this, @JacobHearst. We require a couple of changes before we can merge this PR.

We currently have three different locations where we create the directory of the cache directory

https://github.com/getsentry/sentry-cocoa/blob/f81c2c4bdabdf522c8c6c37d86dd9c26effc2f4c/Sources/SentryCrash/Recording/SentryCrash.m#L76-L92

https://github.com/getsentry/sentry-cocoa/blob/f81c2c4bdabdf522c8c6c37d86dd9c26effc2f4c/Sources/Sentry/SentryFileManager.m#L642-L645

https://github.com/getsentry/sentry-cocoa/blob/f81c2c4bdabdf522c8c6c37d86dd9c26effc2f4c/Sources/Sentry/SentryInstallation.m#L19-L23

To make this work, all three locations need to use the cache directory of the options. As `SentryInstallation id` is static and we don't want to change the signature of the method depend on the options, we could pass the cache dir of the options to `SentryInstallation id` when starting the SDK.

**Comment by philipphofmann** [comment]
Sorry @JacobHearst, for not getting to this by now. Will have a look next week.

**Comment by JacobHearst** [comment]
> Sorry @JacobHearst, for not getting to this by now. Will have a look next week.

No worries, holidays are busy. Happy new year!

**Comment by JacobHearst** [comment]
Hi @philipphofmann and @armcknight my product owner for now just wants our company fork to have the changes we need so that we can use the latest version of Sentry. I'm going to get the changes moved over to the company repo and then open another PR from there. Hopefully this doesn't cause any issues

**Comment by philipphofmann** [comment]
No, that's not a problem, @JacobHearst. Just open your PR again once you are done. Sorry that I didn't get to your PR yet. I was busy preparing the 8.0.0 release.

**Pull request closed without merging**